### PR TITLE
[No Review][5/n] Avoid using hardcoded test credentials

### PR DESCRIPTION
**Pull request by ZhenLian**
This is 5th part of the PR to clean up the test credentials used in C Core(more specifically, in `test/core/end2end/...`).
The ultimate goal is to remove the dependencies on `test/core/end2end/data/ssl_test_data.h`, which contains strings of hard-coded certificates.

### CODE REVIEW

File: test/cpp/util/grpc_tool_test.cc [hunk 1]
```diff
@@ -32,14 +30,21 @@
...
 #include "src/core/lib/gpr/env.h"
+#include "src/core/lib/iomgr/load_file.h"
 #include "src/proto/grpc/testing/echo.grpc.pb.h"
 #include "src/proto/grpc/testing/echo.pb.h"
-#include "test/core/end2end/data/ssl_test_data.h"
 #include "test/core/util/port.h"
 #include "test/core/util/test_config.h"
 #include "test/cpp/util/cli_credentials.h"
 #include "test/cpp/util/string_ref_helper.h"
+#define CA_CERT_PATH "src/core/tsi/test_creds/ca.pem"
```
Comment (jiangtaoli2016) [inline test/cpp/util/grpc_tool_test.cc:43]: empty line between #include and #define

File: test/cpp/util/grpc_tool_test.cc [hunk 2]
```diff
@@ -32,14 +30,21 @@
...
+#define CLIENT_CERT_PATH "src/core/tsi/test_creds/client.pem"
```
Comment (jiangtaoli2016) [inline test/cpp/util/grpc_tool_test.cc:44]: You don't need CA_CERT_PATH, CLIENT_KEY_PATH, CLIENT_CERT_PATH

File: test/cpp/util/test_credentials_provider.cc [hunk 3]
```diff
@@ -19,19 +19,23 @@
...
 #include <mutex>
 #include <unordered_map>
 
-#include <gflags/gflags.h>
-#include <grpc/support/log.h>
-#include <grpc/support/sync.h>
-#include <grpcpp/security/server_credentials.h>
-
-#include "test/core/end2end/data/ssl_test_data.h"
+#include "src/core/lib/iomgr/load_file.h"
```
Comment (jiangtaoli2016) [inline test/cpp/util/test_credentials_provider.cc:33]: It seems that you did not run tools/dockerfile/grpc_clang_format/clang_format_all_the_things.sh to format your change. 
Run
`CLANG_FORMAT_ROOT=~/grpc tools/dockerfile/grpc_clang_format/clang_format_all_the_things.sh`

File: test/cpp/util/test_credentials_provider.cc [hunk 4]
```diff
@@ -19,19 +19,23 @@
...
+#define CA_CERT_PATH "src/core/tsi/test_creds/ca.pem"
```
Comment (jiangtaoli2016) [inline test/cpp/util/test_credentials_provider.cc:34]: Please only use the PATHs that you need.

**Comment by ZhenLian** [comment]
@jiangtaoli2016 this PR keeps failing(for three times) on the test `Bazel C/C++ Dbg MacOS` and `Bazel C/C++ Opt MacOS`, with errors:

> E0415 01:18:52.256368000 140735588610944 tls_pthread.cc:26]            assertion failed: 0 == pthread_setspecific(tls->key, (void*)value)

It looks like this issue is very similar to a known flaky issue https://github.com/grpc/grpc/issues/20311. However, I have ran the tests for 3 times and it all failed on the same issue. I am wondering if I introduced some changes that made this flaky issue failing every time. I might need to dig more on this issue.

**Pull request closed without merging**